File: src/main/java/com/clearcont/clearcontapp/model/Responsavel.java
package com.clearcont.clearcontapp.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
public class Responsavel {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nome;
    @Email
    private String email;
    @ManyToOne
    private Empresa empresa;
    @Setter
    @OneToOne(mappedBy = "responsavel", orphanRemoval = true)
    private User user;

    public Responsavel(Long id, String nome, String email, Empresa empresa) {
        this.id = id;
        this.nome = nome;
        this.email = email;
        this.empresa = empresa;
    }
}

File: src/main/kotlin/br/com/clearcont/clearcontwebapp/models/Responsavel.kt
package br.com.clearcont.clearcontwebapp.models

import jakarta.persistence.*
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotNull

@Entity
class Responsavel() {
    constructor(nome: String, email: String, empresa: Empresa) : this(){
        this.nome = nome
        this.email = email
        this.empresa = empresa
    }

    constructor(email: String, user: ApplicationUser, empresa: Empresa) : this(){
        this.email = email
        this.applicationUser = user
        this.empresa = empresa
    }

    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    var id: Long = 0

    @NotNull
    @Column(length = 50)
    lateinit var nome: String
    @NotNull
    @Email
    @Column(length = 50)
    lateinit var email: String

    @ManyToOne
    lateinit var empresa: Empresa

    @OneToOne(mappedBy = "responsavel", orphanRemoval = true)
    lateinit var applicationUser: ApplicationUser

}
